### PR TITLE
Fix unit status

### PR DESCRIPTION
**Pull request by mykola-kobets-epam**

### CODE REVIEW

File: storagestate/storagestate.go [hunk 1]
```diff
@@ -408,6 +408,9 @@ func (storageState *StorageState) initStateWatching() error {
 	}
 
 	for _, info := range infos {
+		if info.StateQuota == 0 {
+			continue
+		}
```
Comment (al1img) [inline storagestate/storagestate.go:413]: add empy line betwee if

File: unitstatushandler/unitstatushandler.go [hunk 2]
```diff
@@ -175,6 +175,12 @@ func New(
 		systemQuotaAlertChannel: systemQuotaAlertProvider.GetSystemQuoteAlertChannel(),
 	}
 
+	// Initialize containers in order to avoid providing nils to cloud
+	instance.unitStatus.Components = make([]cloudprotocol.ComponentStatus, 0)
```
Comment (al1img) [inline unitstatushandler/unitstatushandler.go:179]: Add all these into `resetUnitStatus` function and call this function here.

File: unitstatushandler/unitstatushandler.go [hunk 3]
```diff
@@ -257,6 +263,9 @@ func (instance *Instance) ProcessRunStatus(instances []cloudprotocol.InstanceSta
 
 	instance.initDone = true
 
+	instance.unitStatus.Instances = instances
```
Comment (al1img) [inline unitstatushandler/unitstatushandler.go:266]: This is not needed as you send zero array from launcher.

File: unitstatushandler/unitstatushandler.go [hunk 4]
```diff
@@ -257,6 +263,9 @@ func (instance *Instance) ProcessRunStatus(instances []cloudprotocol.InstanceSta
...
+	instance.unitStatus.UnitSubjects = make([]string, 0)
```
Comment (al1img) [inline unitstatushandler/unitstatushandler.go:267]: add this to `resetUnitStatus` function as well.

File: unitstatushandler/unitstatushandler.go (hunk 2 above)
Comment (mykola-kobets-epam) [reply unitstatushandler/unitstatushandler.go:179]: done

File: storagestate/storagestate.go (hunk 1 above)
Comment (mykola-kobets-epam) [reply storagestate/storagestate.go:413]: done

File: unitstatushandler/unitstatushandler.go (hunk 3 above)
Comment (mykola-kobets-epam) [reply unitstatushandler/unitstatushandler.go:266]: done

**Review by al1img — APPROVED** [review]

**Review by MykolaSolyanko — APPROVED** [review]

**Review by mlohvynenko — APPROVED** [review]

**Pull request merged**